### PR TITLE
Added waypoint visualization to rendering.py for path planning

### PR DESCRIPTION
**Pull request by wesleyyee1996**
Added waypoint visualization to rendering.py for path planning

Modified: rendering.py, f110_env.py

### CODE REVIEW

**Comment by BDEvan5** [comment]
This looks like a very useful feature, thank you very much.

**Review by hzheng40 — CHANGES_REQUESTED** [review]
I would leave the f110_env.py file unchanged. It's unnecessary for the waypoints to go through that file. The rendering of waypoints should be more of a util function of rendering.py.
By default, render() of f110_env should not render the waypoints. And it should be up to the users to call something other than env.render() to render the waypoints.

File: gym/f110_gym/envs/rendering.py [hunk 1]
```diff
@@ -62,10 +62,14 @@ def __init__(self, width, height, *args, **kwargs):
         Returns:
             None
         """
-        conf = Config(sample_buffers=1,
-                      samples=4,
-                      depth_size=16,
-                      double_buffer=True)
+        
+        # automatically set correct config
```
Comment (hzheng40) [inline gym/f110_gym/envs/rendering.py:66]: The old default shouldn't be deleted. Should probably be a try-except here.

File: gym/f110_gym/envs/rendering.py [hunk 2]
```diff
@@ -294,6 +301,42 @@ def on_draw(self):
...
+        Args:
+            path (array): list of waypoints from path planner
+            
+        Returns:
+            None
+        """
+        
+        # instantiate path w/ set vertex count as Pyglet does not like changing the 
+        # number of vertices for a given polygon after setting it
+        orig_path = [0.]*20
```
Comment (hzheng40) [inline gym/f110_gym/envs/rendering.py:318]: The number of points should probably be an argument of the function

File: gym/f110_gym/envs/rendering.py [hunk 3]
```diff
@@ -294,6 +301,42 @@ def on_draw(self):
...
+        
+        if path is not None:
+            np_path = np.array(path)
+            np_path = np_path.flatten() 
+            
+            # add all points from path to orig_path and set leftover x,y values to
+            # be the final value of path
+            for i in range(0,len(orig_path),2):
+                if i >= len(np_path):
+                    orig_path[i] = 50.*np_path[-2]
```
Comment (hzheng40) [inline gym/f110_gym/envs/rendering.py:328]: Why 50?
Some comments here will be helpful.

File: gym/f110_gym/envs/rendering.py [hunk 4]
```diff
@@ -294,6 +301,42 @@ def on_draw(self):
         self.fps_display.draw()
         # Remove default modelview matrix
         glPopMatrix()
+        
+    def update_waypoints(self, path):
+        """
+        Updates the renderer with the latest waypoint data from the gym environment for the ego vehicle
+        
+        Args:
+            path (array): list of waypoints from path planner
```
Comment (hzheng40) [inline gym/f110_gym/envs/rendering.py:310]: What's the shape requirement?

**Review by hzheng40 — CHANGES_REQUESTED** [review]

File: gym/f110_gym/envs/rendering.py (hunk 3 above)
Comment (hzheng40) [reply gym/f110_gym/envs/rendering.py:328]: My bad, I remember the 50 now, it was used as a scaling factor for rendering. I'll add the comment myself.

**Comment by hzheng40** [comment]
Closing due to inactivity

**Pull request closed without merging**